### PR TITLE
Use Collectible Information for searching items

### LINKED ISSUE #3031: Collections-driven search filters
I'm collecting a bunch of our issues into this one, which is to use the new Collections data to implement new searches:

* https://github.com/DestinyItemManager/DIM/issues/2988 is:year1, is:year2, not:year1, is:season1, is:season2 (year may be inferrable for now without collections)
* https://github.com/DestinyItemManager/DIM/issues/2962 raid sources
* https://github.com/DestinyItemManager/DIM/issues/2249 factions

### PR DESCRIPTION
**Pull request by delphiactual**
~~sourced via https://github.com/delphiactual/d2-seasons/blob/master/generate-engram-rewards.js~~

adds item sources directly to items to allow searching

This is an attempt @ #3031. ~~without using the collection information.~~

Closes #3031

### CODE REVIEW

**Comment by bhollis** [comment]
Pretty cool! Why not go for the collections info though? Items now have a property on them that references the collections they're a part of...

**Comment by delphiactual** [comment]
Honestly because I could not find the property that I was looking for on the items.

**Comment by bhollis** [comment]
Check out `itemDef.collectibleHash`

**Comment by bhollis** [comment]
I'll also mention that the d2-seasons.json file takes about half a megabyte of memory - having these extra databases is sort of expensive.

**Comment by delphiactual** [comment]
changed to using collectibleHash and updated the search.ts but now I am not seeing the filtering working at all

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,12 @@
 * Return mods to the compare view.
 * Item popup background now indicates rarity rather than burn type.
 * The Progress page now shows progress towards reset.
+* Added some vendors to the search dialog.
+    * edz, arcology, nessus, io, mercury, mars
```
[the remark this answers is not in the document]
Comment (kyleshay) [reply docs/CHANGELOG.md]: what if we changed the prefix to `source:` so it was `source:FWC` `source:titan`? The is prefix is so overloaded. Very powerful, but I wonder if we're just digging ourselves deeper. Loving the new search filters though. Amazing what can be done with them.

**Pull request merged**